answers.md

```diff
@@ -0,0 +1,17 @@
+## Part C: Breaking Rc
+
+1. We could break `MyRc` by returning an invalid pointer (i.e. a null pointer) inside of `deref` instead of the real data.
+2. We could break `MyRc` by not checking the count in `drop`; then if we make a copy and drop that, the original will be as well.
+
+Both of these can be seen from almost any normalusage of `MyRc`; we'll almost always want to dereference it, and `drop` is usually desirable, even if it isn't strictly required to be run by Rust. The compiling implementation can be found in `lib.rs` as `MyBrokenRc` and `MyBrokenWeak`; tests that fail can be found in `requiredBroken.rs` and `weakBroken.rs`. 
+
+## Bonus: Unsafe `replace_with`
+The problem is that if a `!panic` occurs, `replace_with` will exit without cleaning things up properly. This results in a situation with unstable values. This can be seen in `replace_with/tests/required.rs`, in the test `safe_memory_error`. Given that we can't recover from an arbitrary `!panic`, and that for something to be memory safe it has to be safe through a `!panic`, it is impossible to implement `replace_with` safely.
```

Comment:
This has the right idea (and gets the bonus points) but is a bit vague. Specifically, during unwinding (what happens after panic), the destructor for `T` **will** be run twice. What exactly this means depend on the `T` and the implementation of `replace_with`, but is likely a problem, and can probably not be guaranteed to be safe for all `T`.